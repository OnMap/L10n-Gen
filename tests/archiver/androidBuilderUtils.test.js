const { convertObjectToParsedXML, buildTranslationsXML } = require('../../src/archiver/utils/androidBuilderUtils');

describe('Builder utils for Android', () => {
  const translationsObject = {
    'activity.filter.allPropertyPresentedOnTvProperties': 'הכל',
    'activity.filter.property.count.show': 'הצג',
    'activity.filter.property.count.properties': 'נכסים'
  };
  const parsedXML = {
    string: [
      {
        _: 'הכל',
        $: { name: 'activity.filter.allPropertyPresentedOnTvProperties' }
      },
      {
        _: 'הצג',
        $: { name: 'activity.filter.property.count.show' }
      },
      {
        _: 'נכסים',
        $: { name: 'activity.filter.property.count.properties' }
      }
    ]
  };

  const xmlString = `<?xml version="1.0" encoding="utf-8"?>
<resources>
  <string name="activity.filter.allPropertyPresentedOnTvProperties">הכל</string>
  <string name="activity.filter.property.count.show">הצג</string>
  <string name="activity.filter.property.count.properties">נכסים</string>
</resources>`;

  it('convert translations object to xmlBuilder format', () => {
    expect(convertObjectToParsedXML(translationsObject)).toEqual(parsedXML);
  });

  it('build XML string from translations object', () => {
    expect(buildTranslationsXML(translationsObject)).toEqual(xmlString);
  });
});
