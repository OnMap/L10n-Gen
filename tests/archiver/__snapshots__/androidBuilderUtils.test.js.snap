// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Builder utils for Android build XML string from translations object 1`] = `
"<?xml version=\\"1.0\\" encoding=\\"utf-8\\"?>
<resources>
  <string name=\\"activity.filter.allPropertyPresentedOnTvProperties\\">הכל</string>
  <string name=\\"activity.filter.property.count.show\\">הצג</string>
  <string name=\\"activity.filter.property.count.properties\\">נכסים</string>
</resources>"
`;

exports[`Builder utils for Android convert translations object to xmlBuilder format 1`] = `
Object {
  "string": Array [
    Object {
      "$": Object {
        "name": "activity.filter.allPropertyPresentedOnTvProperties",
      },
      "_": "הכל",
    },
    Object {
      "$": Object {
        "name": "activity.filter.property.count.show",
      },
      "_": "הצג",
    },
    Object {
      "$": Object {
        "name": "activity.filter.property.count.properties",
      },
      "_": "נכסים",
    },
  ],
}
`;
